### ISSUE TITLE
Metadata values not consistent

### ISSUE THREAD

**Issue by siemvaessen** [IATI/ckanext-iati#270]
**Bug**

For the new Datastore we make use of the IATI Registry metadata. The Field IATI Standard Version prints the version nr. of the publisher's data. Datastore makes use of this version field so it can safely ignore 1.X files seeing Datastore only makes use of 2.X data. Our team noticed that for some (and safe to assume all metadata refresh changes) have an intermediate null state, basically that field has no valid value anymore. Therefore Datastore is led to believe there is no version and therefore is not allowed to pickup the data when it crawls at the time of that intermediate state when no value is presented in the metadata version field. Can this intermediate state in the Registry be valid at all times, ensuring a version number is always printed. This should be applied to any other meta-data field.

**Request**

Ensure all meta data fields in the Registry contain valid values at all times.

![Screenshot 2019-11-11 at 09 42 16](https://user-images.githubusercontent.com/961036/68573731-96b86c00-0468-11ea-8338-0ad1cd6b4dc9.png)

**Comment by andylolz** [comment]
Sounds possibly related to #180.

**Comment by siemvaessen** [comment]
Looks somewhat similar. But that ticket has been closed.

**Comment by andylolz** [comment]
That’s right. I was just flagging it in case there were any useful clues in there.

**Comment by gtkChop** [comment]
@JoshStanleyDevInit @andylolz 

Currently allowed versions are IATI_STANDARD_VERSIONS = [
    '1.01', '1.02', '1.03', '1.04', '1.05', '2.01', '2.02', '2.03'
]. 

Any other versions will have an n/a value. Is this might be a problem. Need to review versions again.

https://github.com/IATI/ckanext-iati/blob/master/ckanext/iati/custom_archiver.py#L275

**Comment by JoshStanleyDevInit** [comment]
@gtkChop Here is an example of a publisher who has just updated their data (within the last hour).

The IATI Standard Version field is set to null and will presumably stay that way until the archiver runs and updates with the actual version of the dataset (2.03 in this case). 

![image](https://user-images.githubusercontent.com/46994072/75872107-89533300-5e05-11ea-9ada-2c42beab26c5.png)

**Comment by JoshStanleyDevInit** [comment]
@gtkChop As discussed, it would be great if the archiver could be automatically run on datasets which have just been updated, thus limiting the time that it spends in a null state awaiting the next time the archiver is run to perhaps a few minutes.

@siemvaessen Would that be preferable from your point of view?

**Comment by siemvaessen** [comment]
Hi @JoshStanleyDevInit - this probably needs a revisit from our side, seeing the new IATI Validator is part of our workflow and we depend on the  IATI Validator summary report (which essentially tells or True/False) prior to parsing/indexing data. @luminhan should be able to tell us.

**Comment by gtkChop** [comment]
@JoshStanleyDevInit

Made archiver run when the dataset has been created/updated. (Please see staging site)

Tests:

- Changed the number of activities to 0/version to 0 and updated the package
Result: Activities and versions are set back to the value from the XML. 

Values set by archiver: 
- IATI Standard Version
- # of Activities

Note: **changes are an almost immediate approx delay of a couple of seconds, Also varies on a number of dataset updated/created**

**Comment by JoshStanleyDevInit** [comment]
@gtkChop I'm struggling to use the staging site at the moment because the pages are not appearing properly.. perhaps something to do with work on #271?

**Comment by gtkChop** [comment]
@JoshStanleyDevInit Yeah, I was working on UI template converting to bootstrap.

Switched the branch again and should be good for your testing. Thanks

**Comment by JoshStanleyDevInit** [comment]
@gtkChop Sorry for the delay on this. I have now tested and am happy for this to be deployed.

Thanks :)
